src/ChannelServer/Skills/Cleric/Heal.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Melia.Channel.Network;
+using Melia.Channel.Skills.Base;
+using Melia.Channel.World.Entities;
+using Melia.Shared.Const;
+using Melia.Shared.Util;
+using Melia.Shared.World;
+
+namespace Melia.Channel.Skills.Cleric
+{
+	/// <summary>
+	/// Handler for the Cleric skill Heal.
+	/// </summary>
+	[SkillHandler(SkillId.Cleric_Heal)]
+	public class Heal : IGroundSkillHandler
+	{
+		/// <summary>
+		/// Handles skill, damaging targets.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="castPosition"></param>
+		/// <param name="targetPosition"></param>
+		public void Handle(Skill skill, Character caster, Position castPosition, Position targetPosition)
+		{
+			// This is just a test for a dedicated skill handler, the code
+			// was largely copied from the default handler for now, without
+			// referencing logs. And the packets are definitely not right
+			// yet.
+
+			var castRange = castPosition.Get3DDistance(targetPosition);
+			if (castRange > skill.Data.MaxRange)
+			{
+				Log.Warning("Heal: Player {0} cast skill farther than max range ({1} > {2}).", caster.Name, castRange, skill.Data.MaxRange);
+				return;
+			}
+
+			if (skill.SpendSp > 0)
+				caster.ModifySp(-skill.SpendSp);
+
+			skill.IncreaseOverheat();
+
+			Send.ZC_SKILL_READY(caster, skill, caster.Position, targetPosition);
+			var healAmount = (int)(caster.GetRandomMagicAmplification() * skill.Data.SkillFactor / 100f);
+
+			Send.ZC_SKILL_MELEE_GROUND(caster, skill, targetPosition, null, 0);
+			// Heal uses ZC_HEAL_INFO instead of ZC_ADD_HP
```

Comment:
Does this cause any issues? Or does `ZC_HEAL_INFO` maybe support any additional features? Should we have a separate method for it?

Reply:
Heal Info seems to be used where visual text is needed for healing, heal spell, recovery (both hp and sp) items. Passive hp/sp recovery is done via ADD_HP/UPDATE_SP.